### PR TITLE
Implement rewind flag in ls and cp commands

### PR DESCRIPTION
**Pull request by vadmeste**
1. --rewind for ls and cp will back in time to select the object versions at that time 
2. ls with --versions will list all versions and will print delete markers with a strike through
3. cp will just ignore if the selected object version is a delete marker

### CODE REVIEW

File: cmd/client.go [hunk 1]
```diff
@@ -61,13 +88,14 @@ type Client interface {
 	SetAccess(ctx context.Context, access string, isJSON bool) *probe.Error
 
 	// I/O operations
-	Copy(ctx context.Context, source string, size int64, progress io.Reader, srcSSE, tgtSSE encrypt.ServerSide, metadata map[string]string, disableMultipart, isPreserve bool) *probe.Error
+	Copy(ctx context.Context, source, versionID string, size int64, progress io.Reader, srcSSE, tgtSSE encrypt.ServerSide, metadata map[string]string, disableMultipart, isPreserve bool) *probe.Error
```
Comment (klauspost) [inline cmd/client.go:91]: Time to have options in a struct?

**Comment by kannappanr** [comment]
Please resolve the conflicts, otherwise LGTM

File: cmd/client.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/client.go:91]: 🍰

Comment (vadmeste) [reply cmd/client.go]: > Time to have options in a struct?

Done

**Pull request merged**